controllers/rack.go

```diff
@@ -848,14 +907,8 @@ func (r *SingleClusterReconciler) rollingRestartRack(
 		podsToRestart = append(podsToRestart, pod)
```

Comment:
This will again append failed pods here as we are already assigning failed pods to `podsToRestart` at line 858 above